zomes/chat/src/error.rs

```diff
@@ -1,10 +1,13 @@
 use hdk3::prelude::*;
+use std::convert::Infallible;
 
 #[derive(thiserror::Error, Debug)]
 pub enum ChatError {
     #[error(transparent)]
     Serialization(#[from] SerializedBytesError),
     #[error(transparent)]
+    Infallible(#[from] Infallible),
```

Comment:
Just wondering what this is needed for? I think by definition this error cannot happen so you probably don't need to return it here. Perhaps somewhere is using `try_from` where it could be just `from`

Reply:
Before adding this type, it was erroring out when trying to compile. Turns out that `emit_signals` expects an implementation of error type Infalliable, which was not included in the ChatResult type.

Here is the error I was seeing at compile time:
![image](https://user-images.githubusercontent.com/30459942/96527957-10470580-1247-11eb-9897-f15e716d08bb.png)